### PR TITLE
Hsharma hackathon

### PR DESCRIPTION
**Pull request by himsharma01**

### CODE REVIEW

File: aws-observability-terraform/source-module/main_iam_role.tf [hunk 1]
```diff
@@ -38,16 +38,16 @@ resource "aws_iam_role_policy_attachment" "cloudtrail_policy_attach" {
 
 # Sumo Logic ELB Source Policy Attachment
 resource "aws_iam_policy" "elb_policy" {
```
Comment (npande) [inline aws-observability-terraform/source-module/main_iam_role.tf:40]: please check the policy is applied to elb and classic lb if buckets are different.

File: aws-observability-terraform/source-module/main_iam_role.tf [hunk 2]
```diff
@@ -38,16 +38,16 @@ resource "aws_iam_role_policy_attachment" "cloudtrail_policy_attach" {
...
+  for_each = toset((local.create_elb_source || local.create_classic_lb_source) && local.create_iam_role ? ["elb_policy"] : [])
   #for_each = toset(var.collect_elb_logs && local.create_iam_role ? ["elb_policy"] : [])
 
   policy = templatefile("${path.module}/templates/iam_s3_source_policy.tmpl", {
-    BUCKET_NAME = local.create_elb_bucket ? local.common_bucket_name : var.elb_source_details.bucket_details.bucket_name
+    BUCKET_NAME = (local.create_elb_bucket || local.create_classic_lb_source) ? local.common_bucket_name : var.elb_source_details.bucket_details.bucket_name
   })
 }
 
 resource "aws_iam_role_policy_attachment" "elb_policy_attach" {
```
Comment (npande) [inline aws-observability-terraform/source-module/main_iam_role.tf:49]: Please verify policy attachment for both elb and classic lb - for different buckets

File: aws-observability-terraform/source-module/outputs.tf [hunk 3]
```diff
@@ -58,6 +58,26 @@ output "elb_auto_enable_stack" {
   description = "AWS CloudFormation stack for ALB Auto Enable access logs."
 }
 
+output "clasic_lb_sns_topic" {
```
Comment (npande) [inline aws-observability-terraform/source-module/outputs.tf:61]: correct spelling for all occurrences - clasic to classic

File: aws-observability-terraform/source-module/outputs.tf [hunk 4]
```diff
@@ -58,6 +58,26 @@ output "elb_auto_enable_stack" {
...
+  value       = local.create_classic_lb_source && !var.classic_lb_source_details.bucket_details.create_bucket ? module.classic_lb_module["classic_lb_module"].aws_sns_topic : {}
+  description = "SNS topic created to be attached to an existing elb logs bucket."
```
Comment (npande) [inline aws-observability-terraform/source-module/outputs.tf:63]: for classic elb outputs occurrences - refer it by classic elb or classic lb

File: aws-observability-terraform/source-module/update_sources.tf [hunk 5]
```diff
@@ -64,6 +64,28 @@ resource "null_resource" "AddFieldsToELBSource" {
...
 }
 
+# Add fields to source if Classic LoadBalancer source already exists
+resource "null_resource" "AddFieldsToCLBSource" {
+   for_each   = toset(local.update_classic_lb_source ? ["add_fields_to_source"] : [])
+   triggers = {
+       access_id         = var.access_id
+       access_key        = var.access_key
+       env               = var.environment
+       source_url        = var.elb_log_source_url
```
Comment (npande) [inline aws-observability-terraform/source-module/update_sources.tf:74]: I presume this should of classic lb - classic_lb_log_source_url

File: aws-observability-terraform/source-module/variables.tf [hunk 6]
```diff
@@ -245,6 +263,44 @@ variable "elb_source_details" {
...
+      force_destroy_bucket = bool
+    })
+    fields = map(string)
+  })
+  description = <<EOT
+            Provide details for the Sumo Logic Classic Load Balancer source. If not provided, then defaults will be used.
+            To enable collection of classic load balancer logs, set collect_classic_lb_logs to true and provide configuration information for the bucket.
+            If create_bucket is false, provide a name of an existing S3 bucket where you would like to store loadbalancer logs. If this is empty, a new bucket will be created in the region.
+            If create_bucket is true, the script creates a bucket, the name of the bucket has to be unique; this is achieved internally by generating a random-id and then post-fixing it to the “aws-observability-” string.
+            path_expression - This is required in case the above existing bucket is already configured to receive ALB access logs. If this is blank, Sumo Logic will store logs in the path expression: *AWSLogs/*/classicloadbalancing/*/*
```
Comment (npande) [inline aws-observability-terraform/source-module/variables.tf:284]: Replace ALB with Classic lb

Comment (himsharma01) [reply aws-observability-terraform/source-module/variables.tf:284]: Done

File: aws-observability-terraform/source-module/update_sources.tf (hunk 5 above)
Comment (himsharma01) [reply aws-observability-terraform/source-module/update_sources.tf:74]: Done

File: aws-observability-terraform/source-module/outputs.tf (hunk 4 above)
Comment (himsharma01) [reply aws-observability-terraform/source-module/outputs.tf:63]: Done

File: aws-observability-terraform/source-module/outputs.tf (hunk 3 above)
Comment (himsharma01) [reply aws-observability-terraform/source-module/outputs.tf:61]: Done

File: aws-observability-terraform/app-modules/elb/app.tf [hunk 7]
```diff
@@ -9,11 +9,13 @@ module "classic_elb_module" {
...
     "ElbAccessLogsFieldExtractionRule" = {
       name             = "AwsObservabilityElbAccessLogsFER"
-      scope            = "account=* region=* namespace=aws/elb"
+      scope            = "account=* region=*"
       parse_expression = <<EOT
         | parse "* * * * * * * * * * * \"*\" \"*\" * *" as datetime, loadbalancername, client, backend, request_processing_time, backend_processing_time, response_processing_time, elb_status_code, backend_status_code, received_bytes, sent_bytes, request, user_agent, ssl_cipher, ssl_protocol
-        | tolowercase(loadbalancername) as loadbalancername
-        | fields loadbalancername
+        | parse regex field=datetime "(?<datetimevalue>\d{0,4}-\d{0,2}-\d{0,2}T\d{0,2}:\d{0,2}:\d{0,2}\.\d+Z)" 
+        | where !isBlank(loadbalancername)and !isBlank(datetimevalue)
```
Comment (npande) [inline aws-observability-terraform/app-modules/elb/app.tf:16]: minor format change, spacing

**Review by npande — APPROVED** [review]
LGTM

**Pull request merged**